### ISSUE TITLE
Add an option to pass a request to the next server on Lua error

### ISSUE THREAD

**Issue by toogle** [tarantool/nginx_upstream_module#26]
`tnt_next_upstream` option supports only few default strategies to pass a request to the next server. In some cases one may need to do it on Lua errors.

Possible solutions are:
1. Return HTTP 500 on Lua errors and use default mask `http_500`.
2. Add a special mask for this case (e.g. `lua_error`).

**Comment by dedok** [comment]
Almost done but before push. I wish to test this feature with balanser_by_lua - from ngx_lua.

Implementation details: 
-- nginx.conf
{
  tnt_next_upstream tnt_error [masks..];
}

**Comment by dedok** [comment]
@toogle Sytax/Option is ok for you. And also I'm interested, did you use balancer_by_lua?

**Comment by toogle** [comment]
No, we didn't. Anyway, it would be great to have such functionality directly in upstream module instead of relying on ngx_lua.

**Comment by zloidemon** [comment]
We also need box.error/error support in protocol, to correct balancing stream in upstream to avoid loss data by tarantool errors.
